File: app/assets/javascripts/MedicinalProductPharmaceutical.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const MedicinalProductPharmaceuticalSchema = DataElementSchema({
  identifier: Any,
  administrableDoseForm: Any,
  unitOfPresentation: Any,
  ingredient: Any,
  device: Any,
  characteristics: Any,
  routeOfAdministration: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'MedicinalProductPharmaceutical' },

});

module.exports.MedicinalProductPharmaceuticalSchema = MedicinalProductPharmaceuticalSchema;
module.exports.MedicinalProductPharmaceutical = mongoose.model('MedicinalProductPharmaceutical', MedicinalProductPharmaceuticalSchema);

File: app/models/qdm/medicinal_product_pharmaceutical.rb
module QDM
  # app/models/qdm/medicinal_product_pharmaceutical.rb
  class MedicinalProductPharmaceutical < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :administrableDoseForm
    field :unitOfPresentation
    field :ingredient
    field :device
    field :characteristics
    field :routeOfAdministration
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
